sound/soc/sof/ipc.c

```diff
@@ -380,7 +395,7 @@ static void ipc_msgs_rx(struct work_struct *work)
 	struct snd_sof_dev *sdev = ipc->sdev;
 	struct sof_ipc_cmd_hdr hdr;
 	u32 cmd, type;
-	int err = -EINVAL;
+	int err = 0;
 
```

Comment:
please be noticed that changing here will consider it as success at case we don't have fw_ready ops.

fw_ready ops is important for us, we need get many information(e.g. memory windows, firmware version, abi version, ...) from it, so I think we need at least throw out some warning logs at it.

Reply:
@ranj063 can you add a warning dev_warn output for the case fw_ready ops is NULL?